### ISSUE TITLE
.gov Registrar: Ensure error icons have accessible names

### ISSUE THREAD

**Issue by SamiyahKey** [cisagov/manage.get.gov#3591]
### Issue description

As a screen reader user, I need error icons in the banner error messages to have alternative text or an accessible label so that I can be informed when an error occurs and understand what action is needed to resolve it.

Current Issue:

- Error indicators are implemented as CSS-based icons without any alternative text or accessible labels (this is different from the inline-field label error messages)

- Screen readers do not announce the presence of the error, preventing users who rely on assistive technologies from being aware of the issue.

URL: In all pages of the .gov Registrar

![Image](https://github.com/user-attachments/assets/1bfbb2f9-0c61-45b0-8c80-639c96245246)

### Acceptance criteria

- [ ] Each error icon has an accessible name that conveys its purpose (e.g., “Error: Invalid email address”).

- [ ] Error icons use aria-label, aria-labelledby for screen reader accessibility.

- [ ] If the error message is visible in text, ensure the icon does not duplicate information but reinforces it programmatically (meaning the error message should not be read out twice).

- [ ] Implementation is verified using assistive technologies such as NVDA, JAWS, or VoiceOver.

### Additional context

This story resolves issue 8 Trusted Tester 508 report and [remediation plan](https://docs.google.com/document/d/1L7TW2RdxAFUrWWLSw3IUYcy4Zu_1OnBl/edit)
